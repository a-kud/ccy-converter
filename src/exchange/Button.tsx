import React from "react";

interface ButtonProps {
  children: React.ReactChild;
  onClick?: () => void;
  className?: string;
}

export function Button({ className, children, onClick }: ButtonProps) {
  const handleClick = () => {
    if (onClick) {
      onClick();
    }
  };

  return (
    <button className={className} disabled={!onClick} onClick={handleClick}>
      {children}
    </button>
  );
}
