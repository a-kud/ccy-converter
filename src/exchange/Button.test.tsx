import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import { Button } from "./Button";

describe("Button", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Button>Nice</Button>);

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
