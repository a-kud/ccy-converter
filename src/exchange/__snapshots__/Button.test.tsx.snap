// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<button
  disabled={true}
  onClick={[Function]}
>
  Nice
</button>
`;
